File: 02-oto-servis/apps/api/Controllers/AuthController.cs
using System.IdentityModel.Tokens.Jwt;
using System.Security.Claims;
using System.Text;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using Microsoft.IdentityModel.Tokens;
using OtoServisApi.Data;

namespace OtoServisApi.Controllers
{
    public class GirisIstegi
    {
        public string Username { get; set; }
        public string Password { get; set; }
    }

    [ApiController]
    [Route("api/auth")]
    public class AuthController : ControllerBase
    {
        private readonly OtoServisContext db;
        private readonly IConfiguration ayarlar;

        public AuthController(OtoServisContext veritabani, IConfiguration konfigurasyon)
        {
            db = veritabani;
            ayarlar = konfigurasyon;
        }

        [HttpPost("login")]
        public async Task<IActionResult> Login(GirisIstegi istek)
        {
            if (string.IsNullOrWhiteSpace(istek.Username) || string.IsNullOrWhiteSpace(istek.Password))
            {
                return BadRequest(new { message = "Kullanıcı adı ve şifre zorunludur." });
            }

            var kullanici = await db.Users
                .FirstOrDefaultAsync(u => u.Username == istek.Username && u.IsActive);

            if (kullanici == null || !BCrypt.Net.BCrypt.Verify(istek.Password, kullanici.PasswordHash))
            {
                return Unauthorized(new { message = "Kullanıcı adı veya şifre hatalı." });
            }

            var haklar = new List<Claim>
            {
                new Claim(ClaimTypes.NameIdentifier, kullanici.Id.ToString()),
                new Claim(ClaimTypes.Name, kullanici.Username),
                new Claim(ClaimTypes.Role, kullanici.Role),
                new Claim("full_name", kullanici.FullName)
            };

            var anahtar = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(ayarlar["Jwt:Key"]));
            var token = new JwtSecurityToken(
                issuer: ayarlar["Jwt:Issuer"],
                claims: haklar,
                expires: DateTime.Now.AddHours(12),
                signingCredentials: new SigningCredentials(anahtar, SecurityAlgorithms.HmacSha256));

            return Ok(new
            {
                token = new JwtSecurityTokenHandler().WriteToken(token),
                user = new
                {
                    id = kullanici.Id,
                    full_name = kullanici.FullName,
                    username = kullanici.Username,
                    role = kullanici.Role
                }
            });
        }

        [HttpGet("me")]
        [Authorize]
        public IActionResult Me()
        {
            return Ok(new
            {
                id = User.FindFirstValue(ClaimTypes.NameIdentifier),
                username = User.FindFirstValue(ClaimTypes.Name),
                role = User.FindFirstValue(ClaimTypes.Role),
                full_name = User.FindFirstValue("full_name")
            });
        }
    }
}

File: 03-klinik-pms/apps/api/src/auth/auth.controller.ts
import {
  BadRequestException, Body, Controller, Get, Post, Req, UnauthorizedException, UseGuards,
} from '@nestjs/common';
import { JwtService } from '@nestjs/jwt';
import { InjectModel } from '@nestjs/mongoose';
import { Model } from 'mongoose';
import * as bcrypt from 'bcryptjs';

import { User, UserDocument } from '../schemas';
import { JwtGuard } from './jwt.guard';

@Controller('api/auth')
export class AuthController {
  constructor(
    @InjectModel(User.name) private kullanicilar: Model<UserDocument>,
    private jwt: JwtService,
  ) {}

  @Post('login')
  async login(@Body() govde: { username: string; password: string }) {
    if (!govde.username || !govde.password) {
      throw new BadRequestException({
        message: 'Kullanıcı adı ve şifre zorunludur.',
      });
    }

    const kullanici = await this.kullanicilar.findOne({
      username: govde.username,
      isActive: true,
    });

    if (!kullanici || !bcrypt.compareSync(govde.password, kullanici.passwordHash)) {
      throw new UnauthorizedException({
        message: 'Kullanıcı adı veya şifre hatalı.',
      });
    }

    const token = await this.jwt.signAsync({
      id: kullanici._id.toString(),
      username: kullanici.username,
      role: kullanici.role,
      full_name: kullanici.fullName,
    });

    return {
      token,
      user: {
        id: kullanici._id.toString(),
        full_name: kullanici.fullName,
        username: kullanici.username,
        role: kullanici.role,
      },
    };
  }

  @Get('me')
  @UseGuards(JwtGuard)
  me(@Req() istek: any) {
    return istek.user;
  }
}
